projects/packages/waf/src/class-waf-runner.php

```diff
@@ -366,24 +375,27 @@ public static function deactivate() {
 
 		// Empty the rules entrypoint file.
 		if ( ! $wp_filesystem->put_contents( self::get_waf_file_path( Waf_Rules_Manager::RULES_ENTRYPOINT_FILE ), "<?php\n" ) ) {
-			throw new \Exception( 'Failed to empty rules.php file.' );
+			throw new File_System_Exception( 'Failed to empty rules.php file.' );
 		}
 	}
 
 	/**
 	 * Handle updates to the WAF
+	 *
+	 * @throws Waf_Exception If updating the WAF fails.
+	 *
+	 * @return void
 	 */
 	public static function update_waf() {
-		Waf_Rules_Manager::update_rules_if_changed();
-		// Re-generate the standalone bootstrap file on every update
-		// TODO: We may consider only doing this when the WAF version changes
 		try {
+			Waf_Rules_Manager::update_rules_if_changed();
+
+			// Re-generate the standalone bootstrap file on every update
+			// TODO: We may consider only doing this when the WAF version changes
 			( new Waf_Standalone_Bootstrap() )->generate();
-		} catch ( \Exception $e ) {
-			return new WP_Error( 'waf_update_failed', $e->getMessage() );
+		} catch ( Waf_Exception $e ) {
+			throw $e;
```

Comment:
Another catch and re-throw.

Reply:
Fixed via https://github.com/Automattic/jetpack/pull/29108/commits/e3edc219c793c772f2411ac87766216aff38f1ae :+1: